cmake/kokkos_arch.cmake

```diff
@@ -144,6 +144,11 @@ ENDIF()
 GLOBAL_SET(KOKKOS_AMDGPU_OPTIONS)
 IF(KOKKOS_CXX_COMPILER_ID STREQUAL HIP)
   SET(AMDGPU_ARCH_FLAG "--amdgpu-target")
+ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL Cray)
```

Comment:
The differentiation in syntax here is really between `hipcc` and `clang`. The options accepted by the Cray compiler for HIP compilation are those accepted by upstream Clang and by the bare ROCm Clang compiler. `hipcc` is just a wrapper that smooths this out.
Would it make more sense to mirror the CUDA Clang logic? That would account for vanilla LLVM builds as well.